packages/esm-form-entry-app/src/app/form-schema/form-schema.service.ts

```diff
@@ -139,28 +172,62 @@ export class FormSchemaService {
     }).pipe(first());
   }
 
-  private fetchFormSchemaUsingFormMetadata(formUuid: string): Observable<any> {
-    return Observable.create((observer: Subject<any>) => {
+  private fetchFormSchemaUsingFormMetadata(
+    formUuid: string,
+    language: string = 'en',
+  ): Observable<FormSchemaAndTranslations> {
+    return Observable.create((observer: Subject<FormSchemaAndTranslations>) => {
       return this.formsResourceService.getFormMetaDataByUuid(formUuid).subscribe(
-        (formMetadataObject: any) => {
+        (formMetadataObject: FormMetadataObject) => {
           if (formMetadataObject.resources.length > 0) {
             const { resources } = formMetadataObject;
-            const valueReferenceUuid = resources?.find(({ name }) => name === 'JSON schema').valueReference;
-            if (valueReferenceUuid) {
-              this.formsResourceService.getFormClobDataByUuid(valueReferenceUuid).subscribe(
-                (clobData: any) => {
-                  observer.next(clobData);
+            const schemaUuid = resources?.find(({ name }) => name === 'JSON schema')?.valueReference;
+            const translationsUuid = resources?.find(({ name }) =>
+              name.endsWith(`_translations_${language}`),
+            )?.valueReference;
+
+            const formSchema = new ReplaySubject<FormSchema>(1);
+            if (schemaUuid) {
+              this.formsResourceService.getFormClobDataByUuid(schemaUuid).subscribe(
+                (clobData) => {
+                  formSchema.next(clobData);
+                },
+                (err) => {
+                  console.error(`Error while loading form ${formUuid}`, err);
+                  formSchema.error(err);
+                },
+              );
+            } else {
+              formSchema.next();
+            }
+
+            const formTranslations = new ReplaySubject<Record<string, string>>(1);
+            if (translationsUuid) {
+              this.formsResourceService.getFormClobDataByUuid(translationsUuid).subscribe(
+                (clobData) => {
+                  formTranslations.next(clobData?.translations ?? {});
                 },
                 (err) => {
-                  console.error(err);
-                  observer.error(err);
+                  console.error(`Error while loading translations for ${formUuid} in ${language}`, err);
+                  formTranslations.error(err);
                 },
               );
             } else {
-              observer.next([]);
+              formTranslations.next({});
             }
+
+            zip(from(formSchema), from(formTranslations))
```

Comment:
Basically, we pair each form schema with a dictionary of translations and return that as an object. I'm not sure it's important, but I tried to ensure there's always a value.